docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,114 @@
+---
+title: Time-Locked Accounts
+hide_title: true
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Time-Locked Accounts
+
+Time-locking allows you to pay someone in MINA or custom other tokens subject to a vesting schedule. This means that the tokens are initially locked, and only become available for withdrawal after a certain time, or gradually according to a certain schedule.
+
+The zkApp feature that enables time-locking is the `timing` field, which is present on every account. It look like this:
+
+```ts
+type Account = {
+  // ...
+  timing: {
+    isTimed: Bool;
+    initialMinimumBalance: UInt64;
+    cliffTime: UInt32;
+    cliffAmount: UInt64;
+    vestingPeriod: UInt32;
+    vestingIncrement: UInt64;
+  };
+};
+```
+
+`isTimed` indicates whether this account is time-locked. The other fields are parameters that allow us to define a vesting schedule in a very flexible manner. We will explain them in detail below. By default, accounts are not time-locked, and `isTimed` will be `false` and all other properties will contain default values.
+
+This graph shows how each of the timing properties affect the vesting schedule:
+
+<figure>
+  <img
+    src="/img/10_time_locked_accounts.png"
+    width="95%"
+    alt="Timing parameters"
+  />
+</figure>
+
+The red cross on the left marks the point in time where the `timing` field is set and `isTimed` switches from `false` to `true`. The blue line shows how the amount of unlocked tokens increases over time, until it finally reaches its maximum value and stays flat. At this point, `isTimed` flips from `true` back to `false`, because no tokens remain locked.
+
+As you can see, the maximum amount of unlocked tokens is defined by the `initialMinimumBalance`. It is called "initialMinimumBalance" because, even though the tokens show up in the balance, they can't be withdrawn — in other words, the account has a a non-zero _minimum balance_. Initially, that minimum balance is equal to the amount of tokens locked (= initial minimum balance). Over time, the minimum balance decreases until it hits zero, which is the condition that makes `isTimed` false again.
```

Comment:
It's not a good time to address this, but `initialMinimumBalance` is an unclear term. 

> (= initial minimum balance)
English please